native/java/org/jpype/JPypeContext.java

```diff
@@ -71,6 +72,9 @@
   private TypeManager typeManager;
   private JPypeReferenceQueue referenceQueue;
   private ClassLoader bootLoader;
+  private AtomicInteger shutdownFlag = new AtomicInteger();
```

Comment:
this gives birth to a beautiful zero? Java flavored obfuscation :D
Later this is checked for being greater zero, so I assume it does.

Reply:
Yep

```
Creates a new AtomicInteger with initial value 0.
```

Everything is born as nothing in Java.  I was a little hesitant to add a counter to the proxies.   Unfortunately a few of my test cases found cases in which I would seg fault if I didn't make sure the proxy was exited.   A full lock system was overkill and potentially rather slow.  So I opted for atomics.